### ISSUE TITLE
ssh到远程服务器，每个指令后都报错warp_preexec:local:1: not valid in this context: . ； 并且打开的文件中文乱码

### ISSUE THREAD

**Issue by MustardWithRice** [warpdotdev/Warp#2971]
### Local Shell and Version

$ZSH_VERSION == 5.8.1

### Remote Shell and Version

$ZSH_VERSION == 5.0.2

### Operating System and Version

centos7

### Warp Version

v0.2023.04.11.08.03.stable_00

### I have tried commenting out my system & user rcfiles and warp still will not bootstrap

- [X] Yes

### Screenshots

warp_preexec:local:1: not valid in this context: . 

�[m~M~U�[m~K�[m~D�[m~P~Fcombo订�[m~M~U�[m~\~I�[m~F�[m~P产�~M~U�[m~Z~D�[m~C~E�[m~F

### Include shell xtrace output

+warptmp.nfJUQT/.zshenv:1> unsetopt ZLE
+warptmp.nfJUQT/.zshenv:1> unset RCS
+warptmp.nfJUQT/.zshenv:1> unset GLOBAL_RCS

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore) - linear-label:e7dfaa84-5fdb-4a00-b754-d8912da923fa

None

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
Thanks for submitting, we'll post any updates on this thread.

You can disable the SSH wrapper by navigating to `Settings > Features`. Blocks will stop working but with it disabled, SSH should work as normal.

Please :+1:  and comment if you're also seeing this issue as it helps us gauge impact.

**Comment by MustardWithRice** [comment]
> 

thanks for your reply。your answer is worked, warp shell is good for developers。I hope you will improve it better and better. If the stability improves, I look forward to experience beyond iterm2。

**Comment by MustardWithRice** [comment]
> Thanks for submitting, we'll post any updates on this thread.
> 
> You can disable the SSH wrapper by navigating to `Settings > Features`. Blocks will stop working but with it disabled, SSH should work as normal.
> 
> Please 👍 and comment if you're also seeing this issue as it helps us gauge impact.

after disabel the SSH wrapper setting, I found another issue:
when I ssh the server and vim a file with Chinese character，some Chinese character don't show, it full display only after sliding the mouse，just like blew screenshot。

<img width="368" alt="class QuotaStore(Resource)" src="https://user-images.githubusercontent.com/38399811/236097539-a44dde98-3ff6-44f0-84b2-fc8177f9e181.png">
<img width="490" alt="class Quotastore(Resource)：" src="https://user-images.githubusercontent.com/38399811/236097568-0d790ca3-b2fe-423c-bd25-d0a584f00557.png">
